staging/src/k8s.io/kubectl/pkg/cmd/wait/wait_test.go

```diff
@@ -799,7 +810,7 @@ func TestWaitForDeletionIgnoreNotFound(t *testing.T) {
 			Namespace: "ns-foo",
 		},
 	}
-	fakeClient := dynamicfakeclient.NewSimpleDynamicClient(scheme)
+	fakeClient := dynamicfakeclient.NewSimpleDynamicClientWithCustomResourceMapping(scheme, listMapping)
 
```

Comment:
is this necessary? could you keep using NewSimpleDynamicClient(scheme) here and in other call sites above?

Reply:
> is this necessary? could you keep using NewSimpleDynamicClient(scheme) here and in other call sites above?

I can't because the existing tests separate resources and versions with unguessable names.  I have to provide the mapping somehow and schemes don't allow it.  I'm not sure how many other locations will face the same issue.  Plurals in english are weird (and that's just english), so I think this is needed to provide people a path to upgrade.

Reply:
ok. the other way is to make it always guessable, which means changing the guess function if/when necessary.

Reply:
> ok. the other way is to make it always guessable, which means changing the guess function if/when necessary.

external components (imagine you have a custom operator) cannot modify that code.